src/analysis/normed_space/lp_space.lean

```diff
@@ -628,6 +628,91 @@ end
 
 end normed_space
 
+section non_unital_normed_ring
+
+variables {I : Type*} {B : I → Type*} [Π i, non_unital_normed_ring (B i)]
+
+lemma _root_.mem_ℓp.infty_mul {f g : Π i, B i} (hf : mem_ℓp f ∞) (hg : mem_ℓp g ∞) : mem_ℓp (f * g) ∞ :=
+begin
+  rw mem_ℓp_infty_iff,
+  obtain ⟨⟨Cf, hCf⟩, ⟨Cg, hCg⟩⟩ := ⟨hf.bdd_above, hg.bdd_above⟩,
+  refine ⟨Cf * Cg, _⟩,
+  rintros _ ⟨i, rfl⟩,
+  calc ∥(f * g) i∥ ≤ ∥f i∥ * ∥g i∥ : norm_mul_le (f i) (g i)
+  ...             ≤ Cf * Cg       : mul_le_mul (hCf ⟨i, rfl⟩) (hCg ⟨i, rfl⟩) (norm_nonneg _)
+                                      ((norm_nonneg _).trans (hCf ⟨i, rfl⟩))
+end
+
+noncomputable instance : has_mul (lp B ∞) :=
+{ mul := λ f g, ⟨λ i, f i * g i, f.property.infty_mul g.property⟩ }
+
+@[simp] lemma lp.infty_coe_fn_mul {f g : lp B ∞} : ⇑(f * g) = f * g := rfl
+
+noncomputable instance : non_unital_ring (lp B ∞) :=
+{ zero_mul := λ f,
+    by { ext, rw [lp.infty_coe_fn_mul, pi.mul_apply, lp.coe_fn_zero, pi.zero_apply, zero_mul] },
+  mul_zero := λ f,
+    by { ext, rw [lp.infty_coe_fn_mul, pi.mul_apply, lp.coe_fn_zero, pi.zero_apply, mul_zero] },
+  mul_assoc := λ f g h, by { ext1, simp only [lp.infty_coe_fn_mul, mul_assoc] },
+  left_distrib := λ f g h,
+    by { ext1, iterate { rw lp.infty_coe_fn_mul <|> rw lp.coe_fn_add }, rw mul_add, },
+  right_distrib := λ f g h,
+    by { ext1, iterate { rw lp.infty_coe_fn_mul <|> rw lp.coe_fn_add }, rw add_mul, },
+  .. lp.normed_group.to_add_comm_group,
+  .. lp.has_mul }
```

Comment:
I think you ought to be able to get this result more concisely with
`function.injective.non_unital_ring`

Reply:
(and for the other norm-less instances below)

Reply:
This works out really nicely for non_unital_ring but I'm unconvinced it saves space for the other instances.